### PR TITLE
Feat/student demo

### PR DESCRIPTION
**Pull request by rojikaru**

### CODE REVIEW

File: Lab2/main.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from models.student import Student
+from models.fee_paying import FeePayingStudent
+
+
+def main():
+    student = Student(1, 'John', 'CS', 1, {'Math': 5, 'History': 4})
+    print(student)
+    print(f'{student.name}\'s grant:', student.grant_size())
```
Comment (dmytro-maherovskyi-lpnu) [inline Lab2/main.py:8]: This print is used several times, can be extracted as a method

File: Lab2/main.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    print('Next course:', student.next_course())
+    print(student)
+
+    print()
+
+    fp_student = FeePayingStudent(2, 'Alice', 'CS', 1, {'Math': 5, 'History': 4}, 1000)
+    print(fp_student)
+    fp_student.deposit(900)
+    print(f'{fp_student.name}\'s balance:', fp_student.get_balance())
```
Comment (dmytro-maherovskyi-lpnu) [inline Lab2/main.py:20]: Same here

File: Lab2/models/student.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        match self.min_score():
+            case 4:
+                return 1000
+            case 5:
+                return 1200
+
+        return 0
+
+    def update_scores(self, course: str, score: int) -> bool:
+        if 1 <= score <= 5:
```
Comment (dmytro-maherovskyi-lpnu) [inline Lab2/models/student.py:45]: Magic numbers should be set as enums or constants

File: Lab2/models/student.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        if self.min_score() > 2:
+            self.year += 1
+            return True
+
+        return False
+
+    def grant_size(self) -> int:
+        match self.min_score():
+            case 4:
+                return 1000
```
Comment (dmytro-maherovskyi-lpnu) [inline Lab2/models/student.py:38]: Magic numbers here as well

**Review by dmytro-maherovskyi-lpnu — CHANGES_REQUESTED** [review]
Good job, but requires minor corrections

File: Lab2/main.py (hunk 1 above)
Comment (rojikaru) [reply Lab2/main.py:7]: Done

File: Lab2/main.py (hunk 2 above)
Comment (rojikaru) [reply Lab2/main.py:29]: Done

File: Lab2/models/student.py (hunk 3 above)
Comment (rojikaru) [reply Lab2/models/student.py]: Done

File: Lab2/models/student.py (hunk 4 above)
Comment (rojikaru) [reply Lab2/models/student.py]: Fixed

**Review by dmytro-maherovskyi-lpnu — APPROVED** [review]

**Pull request merged**